src/realm.js

```diff
@@ -168,6 +169,8 @@ export class Realm {
     this.$GlobalEnv = ((undefined: any): LexicalEnvironment);
 
     this.errorHandler = opts.errorHandler;
+
+    this.globalSymbolRegistry = [];
```

Comment:
@NTillmann It's not great that this moved to the realm since this breaks the spec's notion of a Realm. We'll likely in the future need to have multiple Realms that can interact with each other and transfer objects between each other. Such as with the VM module in the Node interop or iframes in the DOM. In that world, this Symbol table should be shared.

I don't see a reason why this has to be Realm specific instead of a global registry.

Can we revert?

Reply:
An alternative would be to just store this in its own standalone module so that it can be reached from any module.

Reply:
@sebmarkbage:
1. Truly global variables are just bad design.
2. More concretely, I don't like global variables that accumulate stuff as the test runner currently runs all test in one process, just creating and destroying realms. So a truly global registry would be a memory leak, and slow down the test runner.
3. We currently do not actually support multiple realms at the same time, so storing things in the singleton realm is okay.
4. You are right that one day we might want to support multiple Realms as the spec allows, and then the realm-based symbol registry won't work. I should further familiarize myself with the details of how multiple realms are allowed to interact. In any case, the right thing to do seems to me to introduce some new kind of realm-container, we could call it the `JavaScriptEngine` (I guess that's you mean by the "standalone module"), and all of the global state that isn't realm-specific should move in there. That would hopefully reduce our ever-growing Realm class to what it's supposed to be according to the spec, and allow for multiple realms in the future. The engine class would maintain a set of realms, and each realm has a reference back to the containing engine. Does that make sense to you, @sebmarkbage? If so, then I'll create an issue to track this.

Reply:
Yea this can also be instantiated. JS VMs tend to have a concept around this shared thing. The spec doesn't have a name for it since as far as it's concerned it is everything.

V8 calls this an "Isolate". JSC calls it a "VM" I believe. These essentially represent a single shared GC heap. Since they can share references.